### PR TITLE
[16334] Make network headers private

### PR DESCRIPTION
**Pull request by Mario-DL**
Signed-off-by: Mario Dominguez <mariodominguez@eprosima.com>

## Description

Network headers under fastdds/rtps/network/* moved to src/cpp/rtps/network/*
Removed headers in fastrtps/rtps/network/* (those were only pointing to the above ones)

## Contributor Checklist
- [X] Commit messages follow the project guidelines. 
- [X] The code follows the style guidelines of this project. 
**N/A** Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added. 
**N/A** Any new/modified methods have been properly documented using Doxygen. 
- [X] Fast DDS test suite has been run locally. 
- [NO] Changes are ABI compatible. 
- [x] Changes are API compatible. 
**N/A** Documentation builds and tests pass locally. 
- [x] New feature has been added to the `versions.md` file (if applicable).
**N/A** New feature has been documented/Current behavior is correctly described in the documentation. 

## Reviewer Checklist
- [x] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [N/A] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by Mario-DL** [comment]
@richiprosima please test this

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

File: test/unittest/xmlparser/CMakeLists.txt [hunk 1]
```diff
@@ -135,6 +135,7 @@ target_compile_definitions(XMLProfileParserTests PRIVATE FASTRTPS_NO_LIB
     $<$<BOOL:${INTERNAL_DEBUG}>:__INTERNALDEBUG> # Internal debug activated.
     )
 target_include_directories(XMLProfileParserTests PRIVATE
+    ${PROJECT_SOURCE_DIR}/src/cpp
```
Comment (JLBuenoLopez-eProsima) [inline test/unittest/xmlparser/CMakeLists.txt:138]: Why is this change included if the XML Parser tests have not been modified?

File: include/fastrtps/rtps/network/NetworkFactory.h [hunk 2]
```diff
@@ -1,20 +0,0 @@
-// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (JLBuenoLopez-eProsima) [inline include/fastrtps/rtps/network/NetworkFactory.h:1]: Please, be aware that this is not the proper `NetworkFactory` header file but only a "link" to the proper header. You should also modify the `fastdds/rtps/network` folder. Take this into account for the other headers too.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]
There is a couple of methods that are being exported that have as parameter a reference to NetworkFactory. I think that these methods should not be exported so try removing the export macro (`RTPS_DllAPI`) and let us see the Windows CI run. The methods are:

* `ReaderProxyData::readFromCDRMessage`
* `WriterProxyData::readFromCDRMessage`

**Comment by JLBuenoLopez-eProsima** [comment]
On another note, concerning the Contributor checklist:

1. Changes are NOT ABI compatible.
2. These changes should be documented in the versions.md.

File: include/fastdds/rtps/builtin/BuiltinProtocols.h [hunk 3]
```diff
@@ -25,7 +25,7 @@
 
 #include <fastdds/rtps/attributes/RTPSParticipantAttributes.h>
 #include <fastdds/rtps/builtin/data/ContentFilterProperty.hpp>
-#include <fastdds/rtps/network/NetworkFactory.h>
+#include <rtps/network/NetworkFactory.h>
```
[the remark this answers is not in the document]
Comment (Mario-DL) [reply include/fastdds/rtps/builtin/BuiltinProtocols.h:28]: You are right

File: test/unittest/xmlparser/CMakeLists.txt (hunk 1 above)
Comment (Mario-DL) [reply test/unittest/xmlparser/CMakeLists.txt:138]: At some point, it was needed for including dependencies on `<rtps/....>` but since we moved SenderResource.h to `fastdds/...>` it is no longer needed

File: include/fastrtps/rtps/network/NetworkFactory.h (hunk 2 above)
Comment (Mario-DL) [reply include/fastrtps/rtps/network/NetworkFactory.h:1]: According to what we discussed internally, this file should no longer exists here

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test this

File: src/cpp/rtps/builtin/data/ParticipantProxyData.cpp [hunk 4]
```diff
@@ -25,14 +25,14 @@
 #include <fastdds/rtps/builtin/data/WriterProxyData.h>
 #include <fastdds/rtps/builtin/data/ReaderProxyData.h>
 #include <fastdds/rtps/builtin/discovery/participant/PDPSimple.h>
-#include <rtps/network/NetworkFactory.h>
 #include <fastdds/rtps/resources/TimedEvent.h>
 #include <fastrtps/utils/TimeConversion.h>
 
 #include <fastdds/core/policy/ParameterList.hpp>
 #include <fastdds/core/policy/QosPoliciesSerializer.hpp>
 #include <fastrtps_deprecated/participant/ParticipantImpl.h>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/rtps/builtin/data/ParticipantProxyData.cpp:33]: This header file does not seem to be used (and being related to the deprecated API I suggest taking the opportunity and removing it).
Please sort the includes with the recommended order:

* Corresponding header file (ParticipantProxyData.h)
* Blank line
* Standard C header files (there is usually a C++ header file that is preferred so no C header files is advisable).
* Blank line
* Standard C++ files 
* Blank line
* External libraries header files
* Blank line
* Public header files from this library
* Blank line
* Private header files

Each level shall be sorted alphabetically.

This style is not followed in a lot of files, but whenever a header is included or changed, we try to apply this order and try unifying the library code style.

Please, apply in the other files affected by this PR.

File: src/cpp/rtps/network/NetworkFactory.cpp [hunk 5]
```diff
@@ -24,6 +22,7 @@
 #include <fastrtps/utils/IPLocator.h>
 
 #include <rtps/transport/UDPv4Transport.h>
+#include <rtps/network/NetworkFactory.h>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/rtps/network/NetworkFactory.cpp:25]: In this case that it is the corresponding header file, it should be the first one included (even though it is private). See previous comment.

File: src/cpp/rtps/network/ReceiverResource.cpp [hunk 6]
```diff
@@ -12,11 +12,12 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include "ReceiverResource.h"
 #include <fastdds/rtps/messages/MessageReceiver.h>
 #include <cassert>
 #include <fastdds/dds/log/Log.hpp>
 
+#include <rtps/network/ReceiverResource.h>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/rtps/network/ReceiverResource.cpp:19]: This header should be the first to be included.

File: src/cpp/rtps/security/SecurityManager.cpp [hunk 7]
```diff
@@ -48,6 +47,7 @@
 
 #include <rtps/history/TopicPayloadPoolRegistry.hpp>
 #include <rtps/participant/RTPSParticipantImpl.h>
+#include <rtps/network/NetworkFactory.h>
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/rtps/security/SecurityManager.cpp:50]: Please, sort alphabetically (VS Code has a command in the palette: sort lines ascending/descending)

File: versions.md [hunk 8]
```diff
@@ -5,6 +5,7 @@ Forthcoming
 * Added non-standard DataWriterListener callback `on_unacknowledged_sample_removed` (API extension on DDS layer).
 * Added `RTPSWriter::has_been_delivered` virtual method (ABI break on RTPS layer).
 * Refactor `StatefulWriter::get_disable_positive_acks` as virtual method of `RTPSWriter` (ABI break on RTPS layer).
+* Network headers made private.
```
Comment (JLBuenoLopez-eProsima) [inline versions.md:8]:
```suggestion
* Network headers made private (ABI break on RTPS layer).
```

File: src/cpp/rtps/security/SecurityManager.cpp (hunk 7 above)
Comment (Mario-DL) [reply src/cpp/rtps/security/SecurityManager.cpp:50]: Nice to know

File: test/unittest/rtps/builtin/BuiltinDataSerializationTests.cpp [hunk 9]
```diff
@@ -15,11 +15,12 @@
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
+#include <fastdds/core/policy/ParameterSerializer.hpp>
```
Comment (JLBuenoLopez-eProsima) [inline test/unittest/rtps/builtin/BuiltinDataSerializationTests.cpp:18]: Attention: this header is private. It was in the correct section but not alphabetically sorted.

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test windows

**Review by JLBuenoLopez-eProsima — APPROVED** [review]

**Comment by JLBuenoLopez-eProsima** [comment]
Windows CI has to be run so the removal of the export macro (`RTPS_DllAPI`) is checked that does not break anything.

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test windows
